client/v1/thread.js

```diff
@@ -230,16 +230,27 @@ Thread.getById = function (session, id) {
 
 
 Thread.configureText = function(session, users, text) { 
-    if(!_.isArray(users)) users = [users]; 
+    if(!_.isArray(users)) users = [users];
+    var link_urls = Helpers.extractUrl(text);
+    var endpoint = 'threadsBrodcastText';
+
     var payload = {
-        recipient_users: JSON.stringify([users]),
-        client_context: Helpers.generateUUID(),
-        text: text
-    };
+      recipient_users: JSON.stringify([users]),
+      client_context: Helpers.generateUUID()
+    }
+
+    if(link_urls.length > 0) {
+        payload.link.text = text;
```

Comment:
payload.link.text  = text
I'm getting a TypeError: Cannot set property 'text' of undefined

I tried to change it to payload.link_text instead but get 

``` message: 'Not possible to parse API response', ```

Reply:
https://github.com/huttarichard/instagram-private-api/pull/317

Reply:
fixed